### PR TITLE
Migrate Bolt JDBC driver to transaction functions

### LINKED ISSUE #255: Move every session.run into tx-functions
That help Aura users willing to use JDBC

### PR DESCRIPTION
**Pull request by fbiville**
Fixes #255

### CODE REVIEW

**Comment by fbiville** [comment]
The only `session.run` usage left lives in `org.neo4j.jdbc.bolt.SamplePT#testSimpleQueryBoltDriver`, so I guess it's probably fine, @conker84?

File: neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jCallableStatement.java [hunk 1]
```diff
@@ -1,51 +0,0 @@
...
-
-import java.sql.ParameterMetaData;
-import java.sql.ResultSetMetaData;
-import java.sql.SQLException;
-
-/**
- * @author AgileLARUS
- * @since 3.0.0
- */
-abstract class BoltNeo4jCallableStatement extends Neo4jCallableStatement { // TODO not used
```
Comment (conker84) [inline neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jCallableStatement.java:33]: Why you removed this?
Would it be useful for implementing callable statements in future?

**Review by conker84 — APPROVED** [review]
thank you so much @fbiville just a small question otherwise looks good

File: neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jCallableStatement.java (hunk 1 above)
Comment (fbiville) [reply neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jCallableStatement.java:33]: I removed it since this is dead code.

Comment (conker84) [reply neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jCallableStatement.java:33]: Yep but I mean we can use it easily in the implementation of `Connection#prepareCall`

**Pull request merged**